oadp-operator/controllers/restic_test.go

```diff
@@ -0,0 +1,118 @@
+package controllers
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/go-logr/logr"
+	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestVeleroReconciler_ReconcileResticDaemonset(t *testing.T) {
+	type fields struct {
+		Client         client.Client
+		Scheme         *runtime.Scheme
+		Log            logr.Logger
+		Context        context.Context
+		NamespacedName types.NamespacedName
+		EventRecorder  record.EventRecorder
+	}
+	type args struct {
+		log logr.Logger
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		//TODO: Add tests
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r := &VeleroReconciler{
+				Client:         tt.fields.Client,
+				Scheme:         tt.fields.Scheme,
+				Log:            tt.fields.Log,
+				Context:        tt.fields.Context,
+				NamespacedName: tt.fields.NamespacedName,
+				EventRecorder:  tt.fields.EventRecorder,
+			}
+			got, err := r.ReconcileResticDaemonset(tt.args.log)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("VeleroReconciler.ReconcileResticDaemonset() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("VeleroReconciler.ReconcileResticDaemonset() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestVeleroReconciler_buildResticDaemonset(t *testing.T) {
+	type fields struct {
+		Client         client.Client
+		Scheme         *runtime.Scheme
+		Log            logr.Logger
+		Context        context.Context
+		NamespacedName types.NamespacedName
+		EventRecorder  record.EventRecorder
+	}
+	type args struct {
+		velero *oadpv1alpha1.Velero
+		ds     *appsv1.DaemonSet
+	}
+	tests := []struct {
```

Comment:
If you can add just one test that is a valid `velero` and valid `daemonset` object and verify the values match that would be good too